gulpfile.js

```diff
@@ -85,6 +86,84 @@ function createReloadServer() {
   });
 }
 
+function writeServiceWorkerFile() {
+  let path = require('path');
+  let rootDir = 'dist';
+
+  let dynamicUrlToDependencies = {};
+  let partialTemplateFiles = ['head-meta.html', 'site-nav.html']
+    .map(file => path.join(rootDir, 'templates', file));
+
+  ['about', 'index', 'shell'].forEach(htmlFile => {
+    let dynamicUrl = '/' + htmlFile;
+    let dependencies = partialTemplateFiles.concat(`${rootDir}/${htmlFile}.html`);
+    dynamicUrlToDependencies[dynamicUrl] = dependencies;
+    // Treat / like /index
+    if (htmlFile === 'index') {
+      dynamicUrlToDependencies['/'] = dependencies;
+    }
+  });
+
+  let config = {
+    cacheId: 'polymerproject',
+    staticFileGlobs: [
+      `${rootDir}/images/logos/p-logo.png`,
+      `${rootDir}/elements/**`,
+      `${rootDir}/js/*.js`,
+      `${rootDir}/css/*.css`,
+      `${rootDir}/bower_components/**/webcomponents-lite.min.js`,
+      `${rootDir}/bower_components/highlight/highlight.js`,
+    ],
+    dynamicUrlToDependencies: dynamicUrlToDependencies,
+    navigateFallback: '/shell',
```

Comment:
...and just to give some more context about why the App Shell model is something we normally recommend: If you start using individual, full HTML pages cached at runtime to fulfill navigation requests, you run into problems if you ever make changes to your page layout or the list of shared subresources that are referenced in the "boilerplate" HTML.

It's easy to imagine scenarios in which you update your layout, for instance, and then all of your individually cached pages don't get those updates right away. Assuming a stale-while-revalidate (i.e. `fastest`) update strategy, they'll pick up the updates eventually, but only on a URL-by-URL basis. So if you end up with an initial navigation to a URL that was in your runtime cache but was last accessed a year ago, you'll get the layout/subresources from last year, even if you've visited other pages more recently, and that will persist throughout the session due to SPA navigations.

With the App Shell model, there's a single, canonical HTML layout + subresources that is used for all navigations, and as soon as it's updated, all subsequent navigations (both SPA and "real" navigations) will continue to use those updates.

I wrote about this in more detail at https://jeffy.info/2017/01/24/offline-first-for-your-templated-site-part-2.html#messy-updates